tests/test_utils.py

```diff
@@ -4,17 +4,38 @@
 from pathlib import Path
 from unittest.mock import patch
 
+import isceobj  # noqa
 import numpy as np
+import pytest
 from osgeo import gdal
 
 import hyp3_isce2.utils as utils
 
-import isceobj  # noqa
-
 
 gdal.UseExceptions()
 
 
+def test_check_older_granule_is_reference():
+    set1 = ['S1_000000_IW1_20200101T000000_VV_0000-BURST', 'S1_000000_IW1_20200201T000000_VV_0000-BURST']
+    utils.check_older_granule_is_reference(set1[0], set1[1])
+
+    set2 = [
+        ['S1_000000_IW1_20200101T000000_VV_0000-BURST', 'S1_000001_IW1_20200101T000000_VV_0000-BURST'],
+        ['S1_000000_IW1_20200201T000000_VV_0000-BURST', 'S1_000001_IW1_20200201T000000_VV_0000-BURST'],
+    ]
+    utils.check_older_granule_is_reference(set2[0], set2[1])
+
+    set3 = [
+        ['S1_000000_IW1_20200101T000000_VV_0000-BURST', 'S1_000001_IW1_20200101T000000_VV_0000-BURST'],
+        ['S1_000000_IW1_20200201T000000_VV_0000-BURST', 'S1_000001_IW1_20200202T000000_VV_0000-BURST'],
+    ]
+    with pytest.raises(ValueError, match='Reference granules must be from one date*'):
+        utils.check_older_granule_is_reference(set3[0], set3[1])
+
+    with pytest.raises(ValueError, match='Reference granules must be older*'):
+        utils.check_older_granule_is_reference(set1[1], set1[0])
+
+
```

Comment:
This would be easier to read if `check_older_granule_is_reference` were called on the literal values, not defining separate `set1`, `set2`, `set3` variables.

Reply:
Changed!